lib/fastly_nsq/rake_task.rb

```diff
@@ -37,19 +37,16 @@ def run_tasks
       listener_threads = []
 
       topics.each do |topic|
-        listener_threads << Thread.new do
+        thread = Thread.new do
           wrap_helpful_output(topic) do
             MessageQueue::Listener.new(topic: topic, channel: channel).go
           end
         end
+        thread.abort_on_exception = true
```

Comment:
`Thread.abort_on_exception` sets the global state for all threads. As this is a Rake task in a Gem, it seems rather rude to enforce this on all Threads made post this call when it is possible todo at the instance level.

These threads will still cause all others to exit if they raise an exception, but other threads made can act as they choose.